lib/commands/backup.js

```diff
@@ -0,0 +1,231 @@
+        if (folder.indexOf('./') !== 0) {
+            folder = './' + folder;
+        }
+
+        if (ctx.argv.verbose) {
+            ctx.instance.ui.logVerbose(`Backing up ${folder}`,'blue');
+        }
+
+        let files;
+        try {
+            files = walker(folder, {nodir: true});
+        } catch (error) {
+            debug(`Failed reading ${folder}: ${error}`);
+            return Promise.reject(new errors.SystemError({
+                context: error,
+                message: 'Failed to read content folder'
+            }));
+        }
+
+        const cwd = process.cwd();
+
+        const promises = files.map((file) => {
+            const location = file.path.replace(/\\/g,'/');
+            let dir = path.dirname(location);
+            dir = path.relative(cwd, dir);
+            const name = path.basename(location);
+            const localFile = `${dir}/${name}`;
+
+            return fs.readFile(`./${localFile}`).then((fileData) => {
+                ctx.zipFile.addFile(localFile, fileData, '', 644);
+            }).catch((err) => {
+                if (err.code === 'EISDIR')  { // Symlink
+                    return this.backupContent(ctx, null, localFile).catch((error) => {
+                        debug(error);
+                        ctx.instance.ui.log(`Failed to backup ${localFile}`, 'yellow');
+                        // Even though this specific file couldn't be backed up, we want to continue
+                        return Promise.resolve();
+                    });
+                } else {
+                    debug(err);
+                    ctx.instance.ui.log(`Failed to backup ${localFile}`, 'yellow');
+                    // Even though this specific file couldn't be backed up, we want to continue
+                    return Promise.resolve();
+                }
+            });
+        });
+
+        return Promise.all(promises);
+    }
+
+    backupConfig(ctx) {
+        debug('Intializing zip file');
+        try {
+            const configFileName = `config.${ctx.system.environment}.json`;
+            const readFile = fs.readFileSync;
+
+            // Add environment-specific config file
+            ctx.zipFile.addFile(configFileName, readFile(configFileName), '', 644);
+            // Add CLI config file
+            ctx.zipFile.addFile('.ghost-cli', readFile('.ghost-cli'), '', 644);
+
+            return Promise.resolve();
+        } catch (error) {
+            debug(`Failed creating zip: ${error}`);
+            return Promise.reject(new errors.SystemError({
+                context: error,
+                message: 'Failed backing up configuration files'
+            }));
+        }
+    }
+
+    backupExtensions(ctx) {
+        const each = require('lodash/each');
+        const isArray = require('lodash/isArray');
+        const read = fs.readFileSync;
+
+        return ctx.system.hook('backup').then((extensionFiles) => {
+            each(extensionFiles, (filesToAdd) => {
```

Comment:
Just a quick note: you can use `extensionFiles.forEach((filesToAdd) => {})` here, there's not a need for the extra lodash import 😉